python/src/aiconfig/editor/server/server.py

```diff
@@ -566,3 +567,48 @@ def _op(aiconfig_runtime: AIConfigRuntime, _op_args: OpArgs) -> Result[None, str
 
     signature: dict[str, Type[Any]] = {}
     return run_aiconfig_operation_with_request_json(aiconfig, request_json, f"method_", _op, signature)
+
+
+@app.route("/api/get_cli_config", methods=["GET"])
+def get_cli_config() -> FlaskResponse:
+    state = get_server_state(app)
+
+    def _yaml_deserialize(yaml_str: str) -> Result[dict[str, Any], str]:
+        try:
+            out = yaml.safe_load(yaml_str)
+            if out is None:
+                return Err("YAML string not parsed")
+            else:
+                return Ok(out)
+        except Exception as e:
+            return core_utils.ErrWithTraceback(e)
+
+    cli_config_object = core_utils.read_text_file(state.cli_config_path).and_then(_yaml_deserialize)
```

Comment:
Cool is `and_then` another python quirk?

Reply:
nope, it's a Result quirk :)

https://github.com/rustedpy/result

In turn it's basically borrowed from Haskell. [Some of the theory](https://wiki.haskell.org/Monad)

Reply:
lol fun
<img width="418" alt="Screenshot 2024-01-10 at 21 33 50" src="https://github.com/lastmile-ai/aiconfig/assets/151060367/328d17b6-0d73-4941-9a25-94a420cf621c">